lib/utils/to.js

```diff
@@ -29,8 +29,25 @@ module.exports = {
   },
 
   hexWithoutLeadingZeroes: function(val) {
-    val = this.hex(val);
-    return "0x" + val.replace("0x", "").replace(/^0+/, "");
+    val = this.hex(val).replace("0x", "").replace(/^0+/, "");
+    return utils.addHexPrefix(val == "" ? "0" : val);
+  },
+
+  hexWithEvenLength: function(val, width) {
+    val = this.hex(val).replace("0x", "");
+    if (val.length % 2 == 1) {
+      val = "0" + val;
+    }
+    return utils.addHexPrefix(val);
+  },
+
+
+  hexWithLeftPadding: function(val, width) {
```

Comment:
Should this validate the `width` parameter? or at least treat it as 0 when it's falsey?

Reply:
Resolved in f3d3c22

Reply:
Just coming back to this - there are really only two ways prescribed by [the spec](https://github.com/ethereum/wiki/wiki/JSON-RPC#hex-value-encoding) to encode hex values: either as a compact representation with no padding for `QUANTITY` types or as a left-padded representation with two digits per byte for fixed-size `DATA` types.

As such, I feel like the correct fix for padding is to make the object's definition read a bit more like the spec, and then centralize the decision as to how best to pad a particular value. I'll go ahead and rework these changes into a new PR @mjkoo, and if you don't mind, can you please review for me when it's ready?

Reply:
Yeah that makes sense to me, thought about doing that but didn't want to make the patch too intrusive. Sure would be happy to :)